client/my-sites/marketing/connections/service-description.jsx

```diff
@@ -173,8 +173,12 @@ class SharingServiceDescription extends Component {
 				args: { service: this.props.service.label },
 				context: 'Sharing: Publicize',
 			} );
-		} else if ( 'function' === typeof this.props.descriptions[ this.props.service.ID ] ) {
-			description = this.props.descriptions[ this.props.service.ID ].call( this );
+		} else if (
+			'function' === typeof this.props.descriptions[ this.props.service.ID.replace( /-/g, '_' ) ]
```

Comment:
Is there a reason why all the `String.replace` in this PR use a RegExp?

I personally find more readable using a simple string, but maybe the performance is significantly better with a RegExp? 🤔 
```diff
- service.ID.replace( /-/g, '_' )
+ service.ID.replace( '-', '_' )
```

Reply:
Yeah, I found that without the regex it only replaced the first occurrence and so in the case of `instagram-basic-display` the second hyphen wasn't replaced with an underscore.